lab_1/main.py

```diff
@@ -1,94 +1,121 @@
-    presented with the words surrounding it
-    :param tokens: a list of tokens
-    :param word: a word-base for a concordance
-    :param left_context_size: the number of words in the left context
-    :param right_context_size: the number of words in the right context
-    :return: a concordance
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
-                    'the', 'dog', 'is', 'happy', 'but', 'the', 'cat', 'is', 'sad']
-    word = 'happy'
-    left_context_size = 2
-    right_context_size = 3
-    --> [['man', 'is', 'happy', 'the', 'dog', 'is'], ['dog', 'is', 'happy', 'but', 'the', 'cat']]
-    """
-    pass
 
+# получение конкорданса для word
+def get_concordance(tokens: list, word: str, left_n: int, right_n: int) -> list:
+    conco_all = []
+    index_word = 0
+    if checker(tokens, list) and checker(word, str) and checker(left_n, int) and checker(right_n, int):
+        if "".join(tokens) != word:
+            for i in range(tokens.count(word)):
+                if tokens.index(word) == 0:
+                    concordance = tokens[0: index_word + right_n + 1]
+                else:
+                    index_word = tokens.index(word, index_word + 1, len(tokens))
+                    try:
+                        concordance = tokens[index_word - left_n: index_word] + tokens[index_word: index_word + right_context_size + 1]
 
-def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> list:
-    """
-    Gets adjacent words from the left and right context
-    :param tokens: a list of tokens
-    :param word: a word-base for the search
-    :param left_n: the distance between a word and an adjacent one in the left context
-    :param right_n: the distance between a word and an adjacent one in the right context
-    :return: a list of adjacent words
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
-                    'the', 'dog', 'is', 'happy', 'but', 'the', 'cat', 'is', 'sad']
-    word = 'happy'
-    left_n = 2
-    right_n = 3
-    --> [['man', 'is'], ['dog, 'cat']]
-    """
-    pass
+                    except IndexError:
+                        left = index_word - left_n < 0
+                        right = index_word + right_n > len(tokens)
+
+                        if not (left and right):
+                            concordance = tokens[0:len(tokens)]
+                        elif not left and right:
+                            concordance = tokens[0: index_word] + tokens[index_word: index_word + right_n + 1]
+                        elif left and not right:
+                            concordance = tokens[index_word - left_n: index_word] + tokens[index_word: len(tokens)]
+
+                if concordance:
+                    conco_all.append(concordance)
 
+    return conco_all
+
+
+def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> list:
```

Comment:
Please, do not alter the interface of the given functions nor rename the arguments.